src/lib/evil/evil_stat.c

```diff
@@ -0,0 +1,126 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "evil_private.h"
+#include <evil_api.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include "evil_stat.h"
+
+#define	AT_SYMLINK_NOFOLLOW	0x01
+
+EAPI int 
+fstatat(int dirfd, const char *pathname, struct stat *statbuf, int flags)
+{
+   int r_fstatat;
+
+   if (pathname[2] == '\\')
+     {  
+        if (flags == AT_SYMLINK_NOFOLLOW)
+          r_fstatat = stat(pathname, statbuf);
+        else
+          r_fstatat = stat(pathname, statbuf);
+        return r_fstatat;
+     }
+   else
+     {
+        char *pathbuf = NULL;
+        size_t pathbuf_size = 0;
+        DWORD copied = 0;
+        do
+        {
+           pathbuf_size += MAX_PATH;
+           pathbuf = realloc(pathbuf, pathbuf_size * sizeof(char));
+           copied = GetModuleFileName(NULL, pathbuf, pathbuf_size);
+        } while (copied >= pathbuf_size);
+       
+        pathbuf_size = copied;
+        pathbuf = realloc(pathbuf, pathbuf_size * sizeof(char)); 
+        copied = GetModuleFileName(NULL, pathbuf, pathbuf_size);
```

Comment:
Why another call after the loop break?

Reply:
Is it necessary? If I understood the code, it adds nothing.

Reply:
I did a test and removed this repeated code.